packages/payment-processor/test/payment/erc20-batch-proxy.test.ts

```
@@ -0,0 +1,454 @@
+                    ...validRequest.extensions[
+                      PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+                    ],
+                    version: '0.1.0',
+                  },
+                },
+              } as any,
+              {
+                ...validRequest,
+                extensions: {
+                  [PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT]: {
+                    ...validRequest.extensions[
+                      PaymentTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT
+                    ],
+                    version: '0.1.0',
+                  },
+                },
+              } as any,
+            ],
+            batchVersion,
+            batchFee,
+          ).to,
+        ).toBe(batchPaymentsArtifact.getAddress('private', '0.1.0'));
+      });
+    });
+  });
+
+  describe('encodePayBatchRequest: multi tokens', () => {
+    it('should throw an error if the request is not erc20', async () => {
+      const erc20Request = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      const ethRequest = Utils.deepCopy(validRequest) as ClientTypes.IRequestData;
+      ethRequest.currencyInfo.type = RequestLogicTypes.CURRENCY.ETH;
+
+      await expect(
+        payBatchProxyRequest([erc20Request, ethRequest], batchVersion, wallet, batchFee),
+      ).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc20-fee-proxy-contract request',
+      );
+    });
+
+    it('should throw an error if the currencyInfo has no value', async () => {
+      const request = Utils.deepCopy(validRequest);
+      request.currencyInfo.value = '';
+      await expect(
+        payBatchProxyRequest([validRequest, request], batchVersion, wallet, batchFee),
+      ).rejects.toThrowError(
+        'request cannot be processed, or is not an pn-erc20-fee-proxy-contract request',
+      );
+    });
+
+    it('should throw an error if currencyInfo has no network', async () => {
```

You already have this same test in the suite above (I think), in `encodePayErc20BatchRequest`. Same comment for:
- 'should throw an error if the request is not erc20'
- 'should throw an error if the currencyInfo has no value'
- 'should throw an error if request has no extension'
- 'should throw an error if there is a wrong version mapping'

---

updated, now 2 different ERC20